tools/TimeFileMaker.py

```diff
@@ -10,6 +10,20 @@
 STRIP_REP = r'\1'
 INFINITY  = '\xe2\x88\x9e'
 
+def parse_args(argv, USAGE, HELP_STRING):
+    sort_by = 'auto'
+    if any(arg.startswith('--sort-by=') for arg in argv[1:]):
+        sort_by = [arg for arg in argv[1:] if arg.startswith('--sort-by=')][-1][len('--sort-by='):]
+    args = [arg for arg in argv if not arg.startswith('--sort-by=')]
+    if len(args) < 3 or '--help' in args[1:] or '-h' in args[1:] or sort_by not in ('auto', 'absolute', 'diff'):
+        print(USAGE)
+        if '--help' in args[1:] or '-h' in args[1:]:
+            print(HELP_STRING)
+            if len(args) == 2: sys.exit(0)
+        sys.exit(1)
+    return sort_by, args
```

Comment:
Isn't there some cleaner way to do this? In #6519 I used argparse but maybe that's not in some python version you want to support?

Reply:
I am not sure if argparse is in python 2.7 by default.  It also seems a bit heavyweight for handling a single argument like this (which exists anyway only to fix nondeterministic tests).  In any case, I'd rather not put in the effort of switching to argparse, though if these files need another non-positional argument someday, I'd probably make the switch then.  (Feel free to submit a PR switching to argparse or whatever, if you want to, though.)